### ISSUE TITLE
[Feat] Add libdeflate support for accelerated ZIP/Deflate compression/decompression

### ISSUE THREAD

**Issue by neteler** [OSGeo/grass#1046]
**Is your feature request related to a problem? Please describe.**

Inspired by https://github.com/OSGeo/gdal/pull/3068 we may want to look into using "libdeflate": https://github.com/ebiggers/libdeflate ("libdeflate is a library for fast, whole-buffer DEFLATE-based compression and decompression").

For speed comparisons in GDAL, see https://github.com/OSGeo/gdal/pull/3068#issuecomment-710037119

It may well be that this isn't relevant for GRASS GIS but maybe it is...

**Comment by metzm** [comment]
The test results of GDAL are really convincing, but libdeflate is apparently not yet available on recent released Linux distros?

**Comment by neteler** [comment]
That's right and likely the reason that it became an internal library in GDAL as far as I understand.

**Comment by metzm** [comment]
> That's right and likely the reason that it became an internal library in GDAL as far as I understand.

Regarding https://github.com/OSGeo/gdal/pull/3068 it seems that the GDAL internal libtiff can now make use of an external libdeflate. Following GDAL, an external libdeflate could be supported by GRASS, and if available, could be used instead of the standard libzip for (de-)compression.

**Comment by neteler** [comment]
This is what we currently have:
https://grass.osgeo.org/grass-devel/manuals/r.compress.html#overview-of-available-compression-algorithms

Closing.
